From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Wed, 19 Oct 2022 15:24:05 -0700
Subject: [??] wip options as objects instead of interfaces

Flow interfaces can't be spread, or turned into exact object types,
which makes them inconvenient for an options-style object like this.
Use real object types instead.
---
 .../lib/typescript/types.js.flow              | 49 ++++++++++---------
 1 file changed, 25 insertions(+), 24 deletions(-)

diff --git types/@expo/react-native-action-sheet/lib/typescript/types.js.flow types/@expo/react-native-action-sheet/lib/typescript/types.js.flow
index 511fc10f6..266499257 100644
--- types/@expo/react-native-action-sheet/lib/typescript/types.js.flow
+++ types/@expo/react-native-action-sheet/lib/typescript/types.js.flow
@@ -18,28 +18,29 @@ export interface ActionSheetProps {
   ) => void;
 }
 
-export interface ActionSheetIOSOptions {
-  +options: string[];
-  +title?: string;
-  +message?: string;
-  +tintColor?: string;
-  +cancelButtonIndex?: number;
-  +destructiveButtonIndex?: number | number[];
-  +anchor?: number;
-  +userInterfaceStyle?: 'light' | 'dark';
-  +disabledButtonIndices?: number[];
-}
+export type ActionSheetIOSOptions = {|
+  +options: string[],
+  +title?: string,
+  +message?: string,
+  +tintColor?: string,
+  +cancelButtonIndex?: number,
+  +destructiveButtonIndex?: number | number[],
+  +anchor?: number,
+  +userInterfaceStyle?: 'light' | 'dark',
+  +disabledButtonIndices?: number[],
+|};
 
-export interface ActionSheetOptions extends ActionSheetIOSOptions {
-  +icons?: $tsflower_subst$React$ReactNode[];
-  +tintIcons?: boolean;
-  +textStyle?: $tsflower_subst$RN$TextStyle;
-  +titleTextStyle?: $tsflower_subst$RN$TextStyle;
-  +messageTextStyle?: $tsflower_subst$RN$TextStyle;
-  +autoFocus?: boolean;
-  +showSeparators?: boolean;
-  +containerStyle?: $tsflower_subst$RN$ViewStyle;
-  +separatorStyle?: $tsflower_subst$RN$ViewStyle;
-  +useModal?: boolean;
-  +destructiveColor?: string;
-}
+export type ActionSheetOptions = {|
+  ...ActionSheetIOSOptions,
+  +icons?: $tsflower_subst$React$ReactNode[],
+  +tintIcons?: boolean,
+  +textStyle?: $tsflower_subst$RN$TextStyle,
+  +titleTextStyle?: $tsflower_subst$RN$TextStyle,
+  +messageTextStyle?: $tsflower_subst$RN$TextStyle,
+  +autoFocus?: boolean,
+  +showSeparators?: boolean,
+  +containerStyle?: $tsflower_subst$RN$ViewStyle,
+  +separatorStyle?: $tsflower_subst$RN$ViewStyle,
+  +useModal?: boolean,
+  +destructiveColor?: string,
+|};
-- 
2.32.0

